main/networking/Ansible.js

```diff
@@ -185,51 +168,55 @@ class TCPSocket {
       this.logger.log('Runtime disconnected');
     });
 
+    /**
+     * Runtime TCP Message Handler.
+     * TODO: Distinguish between challenge outputs and console logs
+     * when using payload to update console
+     */
     this.socket.on('data', (data) => {
-      const decoded = Notification.decode(data);
-      this.logger.log(`Dawn received TCP Packet ${decoded.header}`);
-
-      switch (decoded.header) {
-        case Notification.Type.CONSOLE_LOGGING:
-          RendererBridge.reduxDispatch(updateConsole(decoded.console_output));
-          break;
-        case Notification.Type.TIMESTAMP_UP:
-          this.logger.log(`TIMESTAMP: ${_.toArray(decoded.timestamps)}`);
-          break;
-      }
+      const decoded = RecvLogProto.decode(data);
+      this.logger.log('Dawn received TCP Packet');
+
+      // TODO: Challenge vs console logs
+      RendererBridge.reduxDispatch(updateConsole(decoded.payload));
     });
 
-    /*
-     * IPC Connection with Editor.js' upload()
-     * When Runtime responds back with confirmation,
-     * notifyChange sends received signal (see tcp, received variables)
+    /**
+     * IPC Connection with sagas.js' exportRunMode()
+     * TODO: Figure out cadence of sending run mode to Runtime
      */
-    ipcMain.on('TIMESTAMP_SEND', this.requestTimestamp);
+    ipcMain.on('EXPORT_RUN_MODE', this.sendRunMode);
   }
 
-  requestTimestamp() {
-    const TIME = Date.now() / 1000.0;
-    const message = Notification.encode(Notification.create({
-      header: Notification.Type.TIMESTAMP_DOWN,
-      timestamps: [TIME],
-    })).finish();
```

Comment:
What were these deleted lines used for previously, just out of curiosity?

Reply:
I think it was a way for Dawn to request timestamps from Runtime. I'm not too sure, but I think it may have been an attempt to build a latency meter (since this stuff has been here since the PM from 3 years ago, and I think he wanted to work on latency detection as his last feature).